conan/api/model.py

```diff
@@ -44,12 +47,13 @@ def add_refs(self, refs):
                 if ref.timestamp:
                     rev_dict["timestamp"] = ref.timestamp
 
-    def add_prefs(self, prefs):
+    def add_prefs(self, rrev, prefs):
         # Prevs already come in ASCENDING order, so upload does older revisions first
+        revs_dict = self.recipes[str(rrev)]["revisions"]
```

Comment:
This guarantees that a "package" section is there, even empty, when package_id patterns are used.